test/new-e2e/tests/agent-shared-components/gui/gui_common.go

```diff
@@ -0,0 +1,185 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package gui
+
+import (
+	"embed"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+	"testing"
+
+	"net/http/cookiejar"
+
+	"github.com/stretchr/testify/assert"
+	"golang.org/x/net/html"
+
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/components"
+)
+
+const (
+	agentAPIPort   = 5001
+	guiPort        = 5002
+	guiAPIEndpoint = "/agent/gui/intent"
+)
+
+//go:embed view
+var expectedAssets embed.FS
+
+// assertAgentsUseKey checks that all agents are using the given key.
+func getGUIIntentToken(t assert.TestingT, host *components.RemoteHost, authtoken string) string {
+	if h, ok := t.(testing.TB); ok {
+		h.Helper()
+	}
+
+	hostHTTPClient := host.NewHTTPClient()
+
+	apiEndpoint := &url.URL{
+		Scheme: "https",
+		Host:   net.JoinHostPort("localhost", strconv.Itoa(agentAPIPort)),
+		Path:   guiAPIEndpoint,
+	}
+
+	req, err := http.NewRequest(http.MethodGet, apiEndpoint.String(), nil)
+	assert.NoErrorf(t, err, "failed to fetch API from %s", apiEndpoint.String())
```

Comment:
Thanks for your comment, you're totally right !
I remembered that it was not recommended to use `require` in `EventuallyWithT` callback function because it might lead to panic.
This behavior have been fixed and the PR to have it in the Agent is currently under review (#31520) (btw thanks @pgimalac for it !).
I've gone ahead and replaced it.